### PR TITLE
Planetary Meteors

### PR DESCRIPTION
**Pull request by Mechoid**
:cl:
adds - New meteor types for waves. (Spalling, Supermatteor)
adds - New meteor types for events (currently). (Meat, Blob)
adds - New meteor type for future in-round triggered events. (Huge)
adds - New rock wall type for Thor, Sif's moon. Presently used in the Huge meteor.
adds - New Easy blob type, used by the Blobteor. Incredibly slow, but slightly more resilient than the Lethargic.
tweak - Meteors can appear planetside, but -everywhere- planetside, that isn't underground.
/:cl:
- [x] Meteors have an angle of attack on planet.
- [x] Meteors can deposit small meteorites
- [x] Meteors can deposit archaeological chunks
- [x] Meteors can be "large", and make a chunk of rock like a drop-pod.

### CODE REVIEW

File: code/modules/blob2/overmind/types/barnacle.dm [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+	brute_multiplier = 0.35
+	burn_multiplier = 0.8
+
+/datum/blob_type/barnacle/on_chunk_use(obj/item/weapon/blobcore_chunk/B, mob/living/user)
+	var/turf/T = get_turf(B)
+
+	to_chat(user, "<span class='alien'>\The [B] produces a cauterizing ooze!</span>")
+
+	T.visible_message("<span class='alium'>\The [B] shudders at \the [user]'s touch, before disgorging a disgusting ooze.</span>")
```
Comment (Atermonera) [inline code/modules/blob2/overmind/types/barnacle.dm:22]: Should you include `usser` in the exclude_mobs list for this call, given the `to_chat()` to them above?
```
/atom/proc/visible_message(var/message, var/blind_message, var/list/exclude_mobs, [...]
```

File: code/modules/blob2/overmind/types/barnacle.dm [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+	for(var/turf/simulated/floor/F in view(2, T))
+		spawn()
```
Comment (Atermonera) [inline code/modules/blob2/overmind/types/barnacle.dm:25]: Is `splash.set_up()` blocking? What's the `spawn()` here for?

File: code/modules/mining/mine_turfs.dm [hunk 3]
```diff
@@ -70,6 +70,24 @@ var/list/mining_overlay_cache = list()
...
 	. = ..()
 
+/turf/simulated/mineral/xenoarch
+	name = "random digsite"
+
+/turf/simulated/mineral/xenoarch/Initialize()
+	. = ..()
+	name = "rock"
+
+	make_digsite(rand(0,2))
```
Comment (Atermonera) [inline code/modules/mining/mine_turfs.dm:80]: `prob(50)`?

File: code/modules/mining/mine_turfs.dm [hunk 4]
```diff
@@ -102,6 +120,30 @@ var/list/mining_overlay_cache = list()
...
+	rock_icon_state = "rock-moon"
+
+/turf/simulated/mineral/moon/xenoarch
+	name = "random thor digsite"
+
+/turf/simulated/mineral/moon/xenoarch/Initialize()
+	. = ..()
+	name = "rock"
+
+	make_digsite(rand(0,2))
```
Comment (Atermonera) [inline code/modules/mining/mine_turfs.dm:136]: As above

File: code/modules/mining/mine_turfs.dm [hunk 5]
```diff
@@ -689,3 +737,42 @@ var/list/mining_overlay_cache = list()
...
+	var/digsite = digsite_type ? digsite_type : get_random_digsite_type()
+
+	for(var/turf/simulated/mineral/T in range(range,src))
+		if(LAZYLEN(T.finds))
+			continue
+
+		if(isnull(T.geologic_data))
+			T.geologic_data = new /datum/geosample(T)
+		
+		if(isnull(T.finds))
```
Comment (Atermonera) [inline code/modules/mining/mine_turfs.dm:754]: This won't populate it with anything if `finds` is an empty `list()`, is that intentional?

File: code/modules/random_map/meteor/meteor.dm [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+	return 1
+
+/datum/random_map/meteor/apply_to_map()
+	if(!applied)
+		applied = TRUE
+		if(placement_explosion_dev || placement_explosion_heavy || placement_explosion_light || placement_explosion_flash)
+			var/turf/T = locate((origin_x + n_ceil(limit_x / 2)-1), (origin_y + n_ceil(limit_y / 2)-1), origin_z)
+			if(istype(T))
+				explosion(T, placement_explosion_dev, placement_explosion_heavy, placement_explosion_light, placement_explosion_flash)
+				sleep(15) // Let the explosion finish proccing before we ChangeTurf(), otherwise it might destroy our spawned objects.
```
Comment (Atermonera) [inline code/modules/random_map/meteor/meteor.dm:78]: There's a spawn _somewhere_ in this callchain, right? We're not just adding a 1.5s sleep to the main processing thread?

File: code/game/gamemodes/meteor/meteors.dm [hunk 7]
```diff
@@ -264,11 +390,41 @@
...
+	adjust_scale(2)
+
+/obj/effect/meteor/huge/meteor_effect(var/explode)
+	..()
+	if(!spawned_terrain)
+		spawned_terrain = TRUE
+		var/dest_x = x - 2
+		var/dest_y = y - 2
+		var/dest_z = z
+		spawn(0)	// Needs to be asynchronous due to how spawning the maps functions.
```
Comment (Atermonera) [inline code/game/gamemodes/meteor/meteors.dm:425]: Ah, excellent.

File: code/modules/random_map/meteor/meteor.dm (hunk 6 above)
Comment (Atermonera) [reply code/modules/random_map/meteor/meteor.dm:78]: (Update: I found the spawn) :+1:

File: code/modules/blob2/overmind/types/barnacle.dm (hunk 2 above)
Comment (Mechoid) [reply code/modules/blob2/overmind/types/barnacle.dm:25]: set_up contains 2 sleeps.

File: code/modules/mining/mine_turfs.dm (hunk 3 above)
Comment (Mechoid) [reply code/modules/mining/mine_turfs.dm:80]: make_digsite's first input is a range, not probability.

**Review by Atermonera — APPROVED** [review]

**Comment by Atermonera** [comment]
Announced via discord at time of update with an @here ping.

**Pull request merged**